slsim/lsst_science_pipeline.py

```diff
@@ -487,3 +498,274 @@ def cutout_image_psf_kernel(
+    return expo_information
+
+
+def list_of_calexp(expo_information, butler):
+    """Extracts calexp images based on exposure information.
+
+    :param expo_information: Astropy table containing exposure information. It must
+        contain Visit ID and Detector ID.
+    :param butler: butler object
+    :return: list of calexp images.
+    """
+    calexp_image = []
+    for visitid, detectorid in zip(
+        expo_information["visitId"], expo_information["detector"]
+    ):
+        dataId = {"visit": visitid, "detector": detectorid}
+        calexp_image.append(butler.get("calexp", dataId=dataId))
+    return calexp_image
+
+
+def warp_to_exposure(exposure, warp_to_exposure):
+    """This function aligns two given dp0 images.
+
+    :param exposure: image that need to be aligned
+    :param warp_to_exposure: Reference image on which exposure should be aligned.
+    :return: Image aligned to reference image.
+    """
+    warper = Warper(warpingKernelName="lanczos4")
+    warped_exposure = warper.warpExposure(
+        warp_to_exposure.getWcs(), exposure, destBBox=warp_to_exposure.getBBox()
+    )
+    return warped_exposure
+
+
+def aligned_calexp(calexp_image):
+    """Alignes list of given images to the first image in the list.
+
+    :param calexp_image: list of calexp images.
+    :return: list of aligned images.
+    """
+    selected_calexp = calexp_image[1:]
+    aligned_calexp_image = [calexp_image[0]]
+    for i in range(0, len(selected_calexp), 1):
+        aligned_calexp_image.append(
+            warp_to_exposure(selected_calexp[i], calexp_image[0])
+        )
+    return aligned_calexp_image
+
+
+def dp0_center_radec(calexp_image):
+    """Computes the center ra, dec of given dp0 image.
+
+    :param calexp_image: dp0 image
+    :return: A sphere point of center ra, dec of given image
+    """
+    bbox = calexp_image.getBBox()
+    xmin, ymin = bbox.getBegin()
+    xmax, ymax = bbox.getEnd()
+    x_center = (xmin + xmax) / 2
+    y_center = (ymin + ymax) / 2
+    radec_point = geom.SpherePoint(calexp_image.getWcs().pixelToSky(x_center, y_center))
+    return radec_point
+
+
+def calexp_cutout(calexp_image, radec, size):
+    """Creates the same size cutouts from given list of dp0 images.
+
+    :param calexp_image: list of dp0 images
+    :param radec: SpherePoint of radec around which we want a cutout
+    :param size: cutout size in pixel unit
+    :return: cutout image of a given size
+    """
+    cutout_extent = geom.ExtentI(size, size)
+    cutout_calexp_image = []
+    for i in range(len(calexp_image)):
+        cutout_calexp_image.append(calexp_image[i].getCutout(radec, cutout_extent))
+    return cutout_calexp_image
+
+
+def radec_to_pix(radec, dp0_image):
+    """Converts ra, dec to pixel units for a fiven image or list of images.
+
+    :param radec: SpherePoint of radec
+    :type radec: an object: eg: geom.SpherePoint(65*degree, -36*degree)
+    :param dp0_image: image containing given radec
```

Comment:
it looks like this can also be a list of images, make sure you write clearly what in and output would look like in multiple images vs single image